packages/central-server/src/apiV2/postChanges.js

```diff
@@ -69,6 +69,8 @@ export async function postChanges(req, res) {
     await ACTION_HANDLERS[action](models, translatedPayload);
 
     if (action === SUBMIT_SURVEY_RESPONSE) {
+      // TODO: Rework this functionality, as directly calling an analytics refresh here is both inefficient
+      // and may create duplicate records in the analytics table
```

Comment:
So the explicit call to `AnalyticsRefresher.refreshAnalytics()` here is dangerous, as it's not wrapped in an advisory lock.

We could consider ways to handle this (eg. move the locking down to the implemented change handler level, use a different mechanism to listen for analytics refreshes, etc.)

However, I sort of wanna call into question if we should be doing this at all...
I know why we did this, to avoid bugs in PSSS where a user would submit data and then wouldn't immediately see it. However, as our system grows in complexity, there are more and more processes that occur asynchronously after data is submitted before it's integrated into the system (analytics refresh, survey response outdate, etc.)
If we wanna support seemless data integration I feel a deeper rework is required to move to an events based system and have the caller listen to some kind of 'data_submission_success' event.

So putting it out there that we just wear this bug for now, if reviewer agrees then I'll put it to Erin 👍